tensorflow/compiler/xla/service/gpu/gpu_executable.cc

```diff
@@ -202,6 +204,9 @@ Status ExecuteThunks(const std::string& module_name,
   tsl::profiler::TraceMe hlo_module_activity(
       [&] { return absl::StrCat(module_name, ":XLA GPU module"); },
       tsl::profiler::TraceMeLevel::kInfo);
+  ScopedAnnotation annotation(
+      [&] { return absl::StrFormat("Thunk:#hlo_module=%s,program_id=%d#",
```

Comment:
The event name should not be "Thunk:", it should be module related message.
Ideally, you could merge this ScopedAnnotation with previous TraceMe with an AnnotatedTraceMe, if they share the same string.

Reply:
I changed "Thunk:" with "XlaModule:". From offline discussion, I was under the impression that this part should be the same to have the 2 scope merged. That was needed to prevent modification to xprof behavior.

Note, I also included a segfault fix. You where right, sometimes no program_id was present. This XLA tests was crashing due to that issue: //tensorflow/compiler/xla/service/gpu/tests:mlir_gemm_test